lib/new_relic/harvest/telemetry_sdk/dimensional_metrics/harvester.ex

```diff
@@ -0,0 +1,164 @@
+    end
+  end
+
+  # do not resend metrics when harvest has already been reported
+  def handle_call(_late_msg, _from, :completed), do: {:reply, :completed, :completed}
+
+  def handle_call(:send_harvest, _from, state) do
+    send_harvest(state)
+    {:reply, :ok, :completed}
+  end
+
+  def handle_call(:gather_harvest, _from, state) do
+    {:reply, build_dimensional_metric_data(state.metrics, state), state}
+  end
+
+  # Helpers
+
+  defp nil_attributes?(attributes) do
+    Map.values(attributes)
+    |> Enum.member?(nil)
+  end
+
+  defp merge_metric(
+         %{type: :summary, name: name, value: new_value, attributes: attributes},
+         metrics_acc
+       ) do
+    new_summary = %{
+      type: :summary,
+      name: name,
+      value: %{
+        count: 1,
+        min: new_value,
+        max: new_value,
+        sum: new_value
+      },
+      attributes: attributes
+    }
+
+    Map.update(
+      metrics_acc,
+      {:summary, name, attributes},
+      new_summary,
+      &update_metric(&1, new_value)
+    )
+  end
+
+  defp merge_metric(
+         %{type: type, name: name, value: new_value, attributes: attributes} = metric,
+         metrics_acc
+       ),
+       do:
+         Map.update(metrics_acc, {type, name, attributes}, metric, &update_metric(&1, new_value))
+
+  defp update_metric(
+         %{type: :count, value: value} = current_metric,
+         new_value
+       ),
+       do: %{current_metric | value: value + new_value}
+
+  defp update_metric(
+         %{type: :gauge} = current_metric,
+         new_value
+       ),
+       do: %{current_metric | value: new_value}
+
+  defp update_metric(
+         %{type: :summary} = current_metric,
+         new_value
+       ),
+       do: %{current_metric | value: update_summary_value_map(current_metric, new_value)}
+
+  defp update_summary_value_map(
+         %{type: :summary, value: value_map},
+         new_value
+       ) do
+    updated_sum_count = %{value_map | sum: value_map.sum + new_value, count: value_map.count + 1}
+
+    updated_min =
+      if new_value < value_map.min,
+        do: %{updated_sum_count | min: new_value},
+        else: updated_sum_count
+
+    if new_value > value_map.max, do: %{updated_min | max: new_value}, else: updated_min
+  end
+
+  defp send_harvest(state) do
+    metrics = Map.values(state.metrics)
+    TelemetrySdk.API.dimensional_metric(build_dimensional_metric_data(metrics, state))
+    log_harvest(length(metrics))
+  end
+
+  defp log_harvest(harvest_size) do
+    NewRelic.log(
+      :debug,
+      "Completed TelemetrySdk.DimensionalMetrics harvest - size: #{harvest_size}"
+    )
+  end
+
+  defp build_dimensional_metric_data(metrics, state) do
+    [
+      %{
+        metrics: metrics,
+        common: common(state)
+      }
+    ]
+  end
+
+  defp common(%{start_time_ms: start_time_ms}) do
+    %{"timestamp" => start_time_ms, "interval.ms" => @interval_ms}
```

Comment:
I'm guessing this doesn't matter, but should interval.ms be a fixed number, or the literal duration between the last harvest cycle and the current harvest?

Here we're saying it's always 5s, whereas our integration in NerdGraph actually computes the delta in case the interval is ever slightly different. Maybe the interval will never drift from 5000ms enough for this to matter

Reply:
If computed, is this what you are thinking?

```suggestion
    %{"timestamp" => start_time_ms, "interval.ms" => System.system_time(:millisecond) - start_time_ms}
```

Reply:
We'd probably want to use monotonic time, I'll send you some code to consider